### ISSUE TITLE
How do I run this?

### ISSUE THREAD

**Issue by sberney** [jvolkman/bazel-nix-python-example#1]
I have a passing familiarity with nix, bazel, and buck. I'm trying to set something up for myself involving some combinations of those things (not opinionated) that will build my python project(s). So far this repo looks like a great starting point, but I'm starting at square 1 with ubuntu. How did you intend this to be used?

I put together this `Dockerfile` that works, but how did you set things up? What's the right way? I don't know what I'm doing, just that I got the following to work, and I'd like to simplify the process. I'm not happy with using docker, for example, but I don't want to clutter my filesystem trying tons of different things either.

```
FROM nixos/nix

RUN nix-env -i bazel

WORKDIR /apps
RUN git clone https://github.com/jvolkman/bazel-nix-python-example.git
WORKDIR bazel-nix-python-example

RUN nix-shell -p openjdk8 --command "bazel build //... && bazel run //app:test_pandas"
```

To run my dockerfile, save it as `Dockerfile` in a new folder `myfolder`. In the `myfolder` directory, run

```
docker build . -t tmp:tmp
```

**Comment by sberney** [comment]
Here is a repo that takes what I posted above and makes it slightly more useful.

https://github.com/sberney/nixbazelpydocker

**Comment by jvolkman** [comment]
With Bazel (or prefereably [Bazelisk](https://github.com/bazelbuild/bazelisk)) and nix installed on your host, this should run as is.
```
bazel test //app/...
```
Did it not work for you?

**Comment by jvolkman** [comment]
Also, if you're interested in getting started with Bazel, consider joining the Bazel slack. I believe I originally created this repo while discussing in the #python channel there, which is likely why there isn't much context in the README here.

https://slack.bazel.build/

**Comment by sberney** [comment]
Thank you for the tips! I'm definitely trying to get more comfortable with bazel.

Yes, it didn't run for me, but that was likely my fault. I wasn't sure if bazel would install nix, or nix would install bazel. So rather than install stuff globally for one example repo I wasn't sure would work, I built the dockerfile. I'm sure it will work when I get my environment set up correctly (fresh OS install).

This is a really excellent example, so thank you. I tried a number of other examples and most of them did not work at all; this was the easiest to succeed with.
